File: RemoveElement.py
class Solution:
    # @param    A       a list of integers
    # @param    elem    an integer, value need to be removed
    # @return an integer
    def removeElement(self, A, elem):
        if A == []:
            return 0

        p = 0
        q = len(A)
        while p < q:
            if A[p] == elem:
                q -= 1
                # swap two elements
                A[p], A[q] = A[q], A[p]
            else:
                p += 1
        return p

if __name__ == "__main__":
    A = [4]
    elem = 4
    Solution().removeElement(A, elem)

File: C++/0/RemoveElement.cc
#include <iostream>
#include <utility>

class Solution {
public:
  int removeElement(int A[], int n, int elem) {
    int removed = 0;
    for (int i = 0; i < n-removed; ++i) {
      if (A[i] == elem) {
        std::swap(A[i], A[n-removed-1]);
        removed++;
        i--;
      }
    }
    return n - removed;
  }
};

int main()
{
  Solution s;
  int A[5] = {1, 2, 2, 2, 2};
  std::cout << s.removeElement(A, 5, 2) << std::endl;
}